### ISSUE TITLE
[UX] Replace Functionality with Features?

### ISSUE THREAD

**Issue by sutibun** [backdrop/backdrop-issues#1150]
Functionality is such a weird long word.

I still remember **that word on the admin bar making me stop, look at it twice, and scratch my head.** Only when I hovered over and saw "modules" did I understand.

How likely are normal users to understand it in relation to modules?

Would "Features" work better? Or maybe

- Extending Features
- Extensions
- Plugins (3rd Party Features)

I wonder if people are more familiar with the word "Features".

- From a salesman, "this product is the best. Let me give you a rundown of all its features"
- "Buy MS Windows because it has the killer feature"
- "What features does this car have?"

Or, could just be me not encountering "functionality" often. Just wondering...

[edit]: Ooopsie. Just saw there was an earlier thread on this topic. Moving comment [there](https://github.com/backdrop/backdrop-issues/issues/209#issuecomment-138133677)

**Comment by quicksketch** [comment]
> [edit]: Ooopsie. Just saw there was an earlier thread on this topic. Moving comment there

Actually let's open this as a separate issue, as the other one is already closed and implemented. As a new change, we should have a new issue.

**Comment by docwilmot** [comment]
Well, since we're reopening, [I completely agree with changing this](https://github.com/backdrop/backdrop-issues/issues/341).

**Comment by sutibun** [comment]
> I truly, completely, do not like the menu item being named "functionality" ... It sounds like something a bunch of engineers in a dark room would be talking about

Agreed. It's terrible like something you'd see in old software developed for MS Windows. The immediate thought to a new user is

> Functionality? What's that? I must be stupid for not knowing what this word means. \*hovers mouse over it\* Modules? WTF? I'm way over my head with this Backdrop thing.

I know from personal experience. My initial impressions of Drupal was that it was hard. Taxonomy. Node. It's like they meant to scare new users away.

If anything, I'd prefer Modules over Functionality, at least, we're being consistent and only scaring them once.

I like the word "Extend". The only problem I see is that it doesn't describe the Modules page it links to well. The modules page lists both default modules (already in core) and contributed modules (where extend would make sense)

That's why I thought **what would best describe a list of both core and contrib modules?** To me, the list of modules defines what Backdrop CAN do. In other words: features. Thus, the Modules listing page is just a list of features. Hence, it would make sense to call it "Features".

But we can take this thinking further. **Core modules are simply core features found in Backdrop by default. Whereas, contrib modules are 3rd party features that extend Backdrop.**

The story aligns better.

**Comment by Graham-72** [comment]
I too like the proposal.
And I think the test of seeing how our terminology works if we use it in sentences is a really good one.

**Comment by sutibun** [comment]
Yay, I was right for once on the Internet :p

**Comment by serundeputy** [comment]
I think `features` is too overloaded a term in the drupal, backdrop and dev space (it has meaning already in development, as a contrib module, now to add as an encompassing menu item seems too much to me).

I can live with `Functionality`.

If we were to change:
  * `Extend` is what D8 uses and it is not bad in my opinion to stay on par vocabulary wise with our Drupal brethren.
  * `Modules` is clear to me.

why is everyone so down on `engineers` there are some engineers here helping, loving, living :do_not_litter: 

~2¢Geoff

**Comment by docwilmot** [comment]
Features is certainly overburdened in Drupal/Backdrop culture.

Extend isn't 100% accurate in terms of core modules, but I suspect most users wouldnt care. I suspect the word 'Extend' would likely be an "ahah, that's gonna be where I get to do new stuff" moment. I don't think a new user would care or even notice that "hey theyve got stuff in here that doesn't literally *extend* the user experience; they lied to me!". 

Functionality is not bad descriptively, but it is too rough for a menu item. Wordpress would never dare put such a word in their admin dashboard.

I think modules is friendlier than functionality, but not as clear. 

I think to satisfy the need for a *friendly* label which is likely to guide the user to where he/she is more likely to be satisfied, "Extend" works a bit better on both counts. It is the better compromise of the bunch.

> why is everyone so down on engineers

:v: Sorry.

**Comment by klonos** [comment]
> I think features is too overloaded a term in the drupal, backdrop and dev space...

> Features is certainly overburdened in Drupal/Backdrop culture.

I feel the same about "features" as a word. It would be difficult to differentiate between the module-feature and the actual feature provided when referenced in READMEs and documentation in general.

**Comment by sutibun** [comment]
Overbloated features
-----

hehe **I didn't even think about the Features module.**

It completely escaped me. Probably because I never picked it up. Think I didn't have enough memory on my shared server or something like that. As a sidenote, not that I fully understand either of them well but isn't the Features module made obsolete with Backdrop's CMI?

Engineer discrimination
----

> I think features is too overloaded a term **in the drupal, backdrop and dev space** (it has meaning already in development, as a contrib module, now to add as an encompassing menu item seems too much **to me**).

& 

> Features is certainly **overburdened in Drupal/Backdrop culture.**

True, it's overloaded for engineers and coders but "Features" is a word normal users can easily grasp.

- "The Features page lists all the **features** you can turn on and off" (*usable*)
- "The Functionality page lists all the **modules** you can turn on and off" (*Has two words that can potentially trip users. Is Functionality even a common word? Sounds like gobbly gook you'd hear from a dry professor working on a research paper ^\_^*)
- "The Module page lists all the **modules** you can turn on and off" (*Way better than functionality, IMO.*)

I have nothing against engineers. At least, I don't think so. I'm coming *from the perspective of normal users.* I'm not saying I myself am a normal user. I'm more like a hybrid. One foot in the dev space and the other in the "normal" realm, mostly because I'm not that smart. One day I'll be more technically capable. One day.

I'm also trying to go by the Backdrop [philosophy](https://backdropcms.org/philosophy)

> Backdrop values **site builders over coders**.
>
> It's most important that the product first be **usable**, and second, easy to extend. **Developers should consider their users first**, and choose to build intuitive user interfaces over flexibility.

Yeah, I threw in the philosophy card :p

Drupal values coders over site builders... maybe you should go ...

hehe just kidding. (^\_-)

The alternative to the alternative
----

*I also like "Extend".*

> Extend isn't 100% accurate in terms of core modules, but I suspect most users wouldnt care. I suspect the word 'Extend' would likely be an "ahah, that's gonna be where I get to do new stuff" moment. I don't think a new user would care or even notice that "hey theyve got stuff in here that doesn't literally extend the user experience; they lied to me!".

Think that'd be the case too.

But here's something else I thought. A compromise of sorts?

Why do we have core modules in the Modules list?

**The core "modules" in Wordpress are so core to the CMS, it's not even listed in the plugins page but are a part of the settings page.** 

- The Wordpress comments "module" can be turned on and off in "Settings > Discussion"
- Want to disable the Wordpress "Path" module, go to "Settings > Permalinks"
- Disable Taxonomy? Just don't turn them on when you create a content type.

Now, that's hardcore :p

Something similar can be done with Backdrop CMS. The Backdrop core modules can finally and truly become core modules by not displaying them on the modules list and instead moving them to the configuration page where you can turn them on and off at will.

Maybe something like http://www.example.com/admin/config/search/clean-urls or you can create a custom section under the configuration listing all the core modules.

Once you remove the core modules from the modules page, you also remove the idea that you can uninstall core modules. (Yeah, I've had that idea early on)

But the best parts are

- there's less scrolling to do in the modules page
- we can now use "Extend" and be Drupal's evil twin brother.

**Comment by jenlampton** [comment]
I'm not actually sure features is to overloaded a word for Backdrop.  For Drupal, yes, but we don't have features in Backdrop (yet?) and we may end up with a different way to solve that problem (bundler module?) since features in Drupal is so often used for config management.

Features is also the most accurate descriptor of what's listed on that page! Modules - those sounds like some growths in my armpits. Ew.  Extend isn't fair, not only does the modules page show what's only already there - but themes and layouts also extend your site!

Features though, that's actually much closer. I like it!

**Comment by mikemccaffrey** [comment]
To me, it seems like part of the problem is that Modules (or whatever you want to call it) should not be a top menu item. Once you've set up a site you rarely have reason to visit that page, and there are never any submenu items. 

Why can't we move the ability manage enabled modules to under the site configuration menu branch?

**Comment by klonos** [comment]
> I'm not actually sure features is to overloaded a word for Backdrop. For Drupal, yes, but we don't have features in Backdrop (yet?) and we may end up with a different way to solve that problem (bundler module?) since features in Drupal is so often used for config management.

Perfect argument there! :+1: ...Now with config in core, I don't think that there will ever be a need for porting [Features](https://www.drupal.org/project/features). So, I believe we are good to "reserve" the word "features" for the modules/functionality page in Backdrop land.

> Features is also the most accurate descriptor of what's listed on that page! Modules - those sounds like some growths in my armpits. Ew.

:+1: :laughing: 

@mikemccaffrey It's not about the UX of people that have finished building their website and simply getting another menu item out of their way so to save space and be minimalistic about the admin bar. It's about the UX of either Drupal/Backdrop newcomers or people coming from other CMSs. So, keeping that in mind, I think that it's good that we keep this as a top-level menu and improve on the terminology used.

**Comment by BWPanda** [comment]
Are we really likely to change the name of this menu item/page at this point? I doubt it, so propose closing this as a "won't fix".

**Comment by jenlampton** [comment]
> Are we really likely to change the name of this menu item/page at this point

Yep! we change the name of menu items all the time. And we change their paths, thanks to `backdrop_goto_deprecated()`. These often lead to major UX wins most of the time :)
